spring-cloud-netflix-eureka-client/src/main/java/org/springframework/cloud/netflix/eureka/EurekaClientAutoConfiguration.java

```diff
@@ -113,38 +113,46 @@ public EurekaClientConfigBean eurekaClientConfigBean() {
 	}
 
 	@Bean
+	public ServiceMetadataProvider eurekaServiceMetadataProvider() {
+		return new DefaultServiceMetadataProvider();
+	}
+
+	@Bean
 	@ConditionalOnMissingBean(value = EurekaInstanceConfig.class, search = SearchStrategy.CURRENT)
-	public EurekaInstanceConfigBean eurekaInstanceConfigBean(InetUtils inetUtils) throws MalformedURLException {
+	public EurekaInstanceConfigBean eurekaInstanceConfigBean(InetUtils inetUtils,
+															 ServiceMetadataProvider serviceMetadataProvider) {
 		PropertyResolver eurekaPropertyResolver = new RelaxedPropertyResolver(this.env, "eureka.instance.");
 		String hostname = eurekaPropertyResolver.getProperty("hostname");
 
 		boolean preferIpAddress = Boolean.parseBoolean(eurekaPropertyResolver.getProperty("preferIpAddress"));
+		String serverContextPath = propertyResolver.getProperty("server.contextPath", "/");
 		int nonSecurePort = Integer.valueOf(propertyResolver.getProperty("server.port", propertyResolver.getProperty("port", "8080")));
-		int managementPort = Integer.valueOf(propertyResolver.getProperty("management.port", String.valueOf(nonSecurePort)));
-		String managementContextPath = propertyResolver.getProperty("management.contextPath", propertyResolver.getProperty("server.contextPath", "/"));
+		Integer managementPort = propertyResolver.getProperty("management.port", Integer.class);// nullable. should be wrapped into optional
+		String managementContextPath = propertyResolver.getProperty("management.contextPath");// nullable. should be wrapped into optional
+
 		EurekaInstanceConfigBean instance = new EurekaInstanceConfigBean(inetUtils);
 		instance.setNonSecurePort(nonSecurePort);
 		instance.setInstanceId(getDefaultInstanceId(propertyResolver));
 		instance.setPreferIpAddress(preferIpAddress);
-		if (managementPort != nonSecurePort && managementPort != 0) {
```

Comment:
Why there was a check for managementPort != nonSecurePort?
Shouldn't we set urls if management port is equal to nonSecurePort?